googleworkspace/table_googleworkspace_admin_reports_activity.go

```diff
@@ -0,0 +1,216 @@
+            },
+            {
+                Name:        "events",
+                Description: "Full JSON array of detailed events (Events)",
+                Type:        proto.ColumnType_JSON,
+                Transform:   transform.FromField("Events"),
+            },
+        },
+    }
+}
+
+//// LIST FUNCTION
+
+func listGoogleworkspaceAdminReportsActivities(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+    service, err := ReportsService(ctx, d)
+    if err != nil {
+        plugin.Logger(ctx).Error("googleworkspace_admin_reports_activity.list", "service_error", err)
+        return nil, err
+    }
+
+    // Required application_name qualifier
+    appName := d.EqualsQualString("application_name")
+
+    // Validate application_name
+    valid := map[string]bool{
+        "access_transparency":          true,
+        "admin":                        true,
+        "calendar":                     true,
+        "chat":                         true,
+        "drive":                        true,
+        "gcp":                          true,
+        "gplus":                        true,
+        "groups":                       true,
+        "groups_enterprise":            true,
+        "jamboard":                     true,
+        "login":                        true,
+        "meet":                         true,
+        "mobile":                       true,
+        "rules":                        true,
+        "saml":                         true,
+        "token":                        true,
+        "user_accounts":                true,
+        "context_aware_access":         true,
+        "chrome":                       true,
+        "data_studio":                  true,
+        "keep":                         true,
+        "vault":                        true,
+        "gemini_in_workspace_apps":     true,
+}
+
+if !valid[appName] {
+    return nil, fmt.Errorf("unsupported application_name: %q", appName)
+}
+
+    // Build API call with the chosen category
+    call := service.Activities.List("all", appName)
+
+    if quals := d.Quals["time"]; quals != nil {
+        var startTime, endTime time.Time
+        for _, q := range quals.Quals {
+            if ts := q.Value.GetTimestampValue(); ts != nil {
+                t := ts.AsTime()
+                switch q.Operator {
+                case "=":
+                    startTime, endTime = t, t
+                case ">":
+                    startTime = t.Add(time.Nanosecond)
+                case ">=":
+                    startTime = t
+                case "<":
+                    endTime = t
+                case "<=":
+                    endTime = t
+                }
+            }
+        }
+        if !startTime.IsZero() {
+            call.StartTime(startTime.Format(time.RFC3339))
+        }
+        if !endTime.IsZero() {
+            call.EndTime(endTime.Format(time.RFC3339))
+        }
+    }
+

```

Comment:
Could you also take care of the remaining optional qualifiers defined in the list config? I noticed they don't appear to be used when making the API call.